include/revng/ADT/Concepts.h

```diff
@@ -32,6 +32,11 @@ concept range = requires(T &R) {
   std::end(R);
 };
 
+template<class Container, typename ElementType>
+concept typed = range<Container> &&
+                std::is_convertible_v<typename Container::value_type,
+                                      ElementType>;
```

Comment:
About this, the `ranges` namespace here was designed to mirror the c++20's ranges, since there's nothing like this one in the standard, I feel like it's better to move it outside the namespace - to avoid confusion when we update the compiler and get rid of the two other ones in favour of the standard ones.

Reply:
(I didn't notice it was in a namespace)

Reply:
Fixed